src/Algebra/Graph/Relation/Symmetric/Internal.hs

```diff
@@ -0,0 +1,193 @@
+--
+-- @
+-- connect x y == connect y x
+-- 'Algebra.Graph.Relation.Symmetric.isEmpty'     (connect x y) == 'Algebra.Graph.Relation.Symmetric.isEmpty'   x   && 'Algebra.Graph.Relation.Symmetric.isEmpty'   y
+-- 'Algebra.Graph.Relation.Symmetric.hasVertex' z (connect x y) == 'Algebra.Graph.Relation.Symmetric.hasVertex' z x || 'Algebra.Graph.Relation.Symmetric.hasVertex' z y
+-- 'Algebra.Graph.Relation.Symmetric.vertexCount' (connect x y) >= 'Algebra.Graph.Relation.Symmetric.vertexCount' x
+-- 'Algebra.Graph.Relation.Symmetric.vertexCount' (connect x y) <= 'Algebra.Graph.Relation.Symmetric.vertexCount' x + 'Algebra.Graph.Relation.Symmetric.vertexCount' y
+-- 'Algebra.Graph.Relation.Symmetric.edgeCount'   (connect x y) >= 'Algebra.Graph.Relation.Symmetric.edgeCount' x
+-- 'Algebra.Graph.Relation.Symmetric.edgeCount'   (connect x y) >= 'Algebra.Graph.Relation.Symmetric.edgeCount' y
+-- 'Algebra.Graph.Relation.Symmetric.edgeCount'   (connect x y) >= 'Algebra.Graph.Relation.Symmetric.vertexCount' x * 'Algebra.Graph.Relation.Symmetric.vertexCount' y `div` 2
+-- 'Algebra.Graph.Relation.Symmetric.vertexCount' (connect 1 2) == 2
+-- 'Algebra.Graph.Relation.Symmetric.edgeCount'   (connect 1 2) == 1
+-- @
+connect :: Ord a => SymmetricRelation a -> SymmetricRelation a -> SymmetricRelation a
+connect (SymmetricRelation x) (SymmetricRelation y) = SymmetricRelation $ Relation 
+                                                      (domain x `Set.union` domain y)
+                                                      (relation x `Set.union` relation y 
+                                                      `Set.union` 
+                                                      (domain x `setProduct` domain y)
+                                                      `Set.union` 
+                                                      (domain y `setProduct` domain x))
+
+instance (Ord a, Num a) => Num (SymmetricRelation a) where
+    fromInteger = vertex . fromInteger
+    x + y       = x `overlay` y
+    x * y       = x `connect` y
+    signum      = const empty
+    abs         = id
+    negate      = id
+
+-- | Check if the internal representation of a relation is consistent, i.e. if all
+-- pairs of elements in the 'relation' refer to existing elements in the 'domain' and that if all pair of edges have their symmetric.
+-- It should be impossible to create an inconsistent 'Relation', and we use this
+-- function in testing.
+-- /Note: this function is for internal use only/.
+--
+-- @
+-- consistent 'Algebra.Graph.Relation.empty'         == True
+-- consistent ('Algebra.Graph.Relation.vertex' x)    == True
+-- consistent ('Algebra.Graph.Relation.overlay' x y) == True
+-- consistent ('Algebra.Graph.Relation.connect' x y) == True
+-- consistent ('Algebra.Graph.Relation.edge' x y)    == True
+-- consistent ('Algebra.Graph.Relation.edges' xs)    == True
+-- consistent ('Algebra.Graph.Relation.stars' xs)    == True
```

Comment:
Responding to your questions in the `compose` thread:

Here is where we claim that various functions in our API are consistent. We could add more such claims, especially for tricky functions like `compose`. And, of course, all such claims should be tested.

Reply:
I added the tests but had to force some of the variable types otherwise the ghc would give errors. I leaved a TODO in the comments for this as well.